src/rules-parser.pegjs

```diff
@@ -140,13 +111,27 @@ PathExpression "path" =  parts:("/" part:PathKey? { return part; })+ {
     return part;
   });
   if (hasError) {
-    error((parts[parts.length - 1] === '' ? "Paths may not end in a slash (/) character"
+    error((parts[parts.length - 1] === ''
+           ? "Paths may not end in a slash (/) character"
            : "Paths may not contain an empty part") + ": /" + parts.join('/'));
   }
-  return parts;
+  return new ast.PathTemplate(parts);
 }
 
-PathKey = chars: [^ /;]+ { return chars.join(''); }
+PathKey = CaptureKey / LiteralPathKey
+
+CaptureKey = "{" _ id:Identifier _ ("=" _ "*" _ )? "}" {
```

Comment:
Firebase cannot handle capturing anything with "/" in it - so those glob
patterns are out.

On Mon, Jan 11, 2016 at 11:20 AM, Tristan Swadell <notifications@github.com>
wrote:

> In src/rules-parser.pegjs
> <https://github.com/firebase/bolt/pull/105#discussion_r49364881>:
>
> >  }
> >
> > -PathKey = chars: [^ /;]+ { return chars.join(''); }
> > +PathKey = CaptureKey / LiteralPathKey
> > +
> > +CaptureKey = "{" _ id:Identifier _ ("=" _ "*" _ )? "}" {
>
> What is the subset of patterns we intend to support on CaptureKeys?
> I'm curious about patterns like the following: {var = **}, {var = */*}, {var
> = */**}
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/firebase/bolt/pull/105/files#r49364881>.
>



-- 
+Mike Koss <https://plus.google.com/+MikeKoss>
(425) 246-7701 (m)